### ISSUE TITLE
The name 'X' does not exist in the current context

### ISSUE THREAD

**Issue by tpluscode** [Fody/Fody#130]
Hi. We've been having two problems with debugging our project, where we use Fody extensively.

First, which we have been living with for a long time, is that in watch/immediate window fully qualified type names must be used even if that. This is especially annoying with extension methods so that

```
System.Linq.Enumerable.Count(collection)
```

must be used instead of simple

```
collection.Count()
```

which fails saying that method `Count` cannot be found. I managed to replicate this and it seem that it's not exactly a PDB problem, because referencing a Fody-modified assembly works fine. The problem only occurs when the currently debugged code is built with Fody,.

The other problem seem to have started occuring very recently. Definitely in 1.23.x. For some local variables Visual Studio 2012 says that they are not in the current context much like in issue #120 though I was unable to replicate the issue just yet.

I've just updated to 1.24 and the problem is still there.

Would you like for me to supply some code to replicate the problems or is there something I could try first?

**Comment by davidroth** [comment]
@jbevain Do you have any progress on this issue? Can we expect a fix anytime soon?

**Comment by SimonCropp** [comment]
can u try 2.0

**Comment by SimonCropp** [comment]
@jbevain in an assembly built in release mode round tripping still results in this problem.  (in the latest master of cecil).

**Comment by SimonCropp** [comment]
@jbevain i note that disabling "Optimize Code" make it work again 
![image](https://cloud.githubusercontent.com/assets/122666/25419723/a917e36a-2a97-11e7-864a-b53765fa972b.png)

**Comment by tpluscode** [comment]
Optimized code has its debugging issues anyway. I'd consider this issue resolved and add a notice in the readme.

**Comment by SimonCropp** [comment]
@tpluscode agreed "Optimized code has its debugging issues anyway". but this is a little more broken than that

**Comment by tpluscode** [comment]
I see. What exactly happens when you try to debug optimized code processed by Fody?

**Comment by SimonCropp** [comment]
u can break point. but most variables cannot be inspected

**Comment by bradphelan** [comment]
Has this

> First, which we have been living with for a long time, is that in watch/immediate window fully qualified type names must be used 

been resolved in either Fody or Cecil yet. It's been an annoyance for at least a year now. Not enough to drop Fody but the immediate window is unusable during debugging. DEBUG or RELEASE mode.

**Comment by bradphelan** [comment]
I just upgraded Fody to the latest nuget package and my problems have gone away.  🥇

**Comment by SimonCropp** [comment]
havnt seen this one in a while so closing

**Comment by ptixed** [comment]
Are you sure this was fixed? I just installed latest version 3.3.5 and am seeing the same behavior. Can use `typeof(MyProject.Program)` but not `typeof(Program)` in debug view. It works fine when no weavers are applied.

**Comment by SimonCropp** [comment]
@ptixed suggest u raise a new issue
